modules/dnn/src/net_impl.cpp

```diff
@@ -186,11 +186,6 @@ void Net::Impl::setUpNet(const std::vector<LayerPin>& blobsToKeep_)
 
         clear();
 
-        if (hasDynamicShapes)
-        {
-            updateLayersShapes();
-        }
-
```

Comment:
Why?

Reply:
As I understand, this function runs shape inference for the second time for the models with dynamic shapes. I've found out that this is redundant because this line runs before the main shape inference.

Reply:
@dkurt could you take a look?